crates/router/src/services/api.rs

```diff
@@ -766,6 +766,7 @@ where
         .switch()
     })?;
     session_state.add_request_id(request_id);
+    session_state.add_header_map(incoming_request_header.clone());
```

Comment:
Ummm, why are we populating _incoming_ request headers as headers for the API client which is used to make _outgoing_ HTTP requests?

Reply:
yes this should not be here, moved to struct `SessionState`